src/lcm/entry_point.py

```diff
@@ -59,196 +63,116 @@ def get_lcm_function(
+            next_state_space_info = LastPeriodsNextStateSpaceInfo
+        else:
+            next_state_space_info = state_space_infos[period + 1]
 
-        # create the compute conditional continuation value functions and append to list
-        # ==============================================================================
         u_and_f = get_utility_and_feasibility_function(
-            model=_mod,
-            state_space_info=state_space_infos[period],
+            model=internal_model,
+            next_state_space_info=next_state_space_info,
             period=period,
             is_last_period=is_last_period,
         )
 
-        compute_ccv = create_compute_conditional_continuation_value(
+        compute_ccv = get_compute_conditional_continuation_value(
             utility_and_feasibility=u_and_f,
-            continuous_choice_variables=list(_choice_grids),
+            continuous_choice_variables=tuple(state_choice_space.continuous_choices),
         )
-        compute_ccv_functions.append(compute_ccv)
 
-        compute_ccv_argmax = create_compute_conditional_continuation_policy(
+        compute_ccp = get_compute_conditional_continuation_policy(
             utility_and_feasibility=u_and_f,
-            continuous_choice_variables=list(_choice_grids),
+            continuous_choice_variables=tuple(state_choice_space.continuous_choices),
         )
-        compute_ccv_policy_functions.append(compute_ccv_argmax)
 
-        # create list of emax_calculators
-        # ==============================================================================
-        calculator = get_solve_discrete_problem(
-            random_utility_shock_type=_mod.random_utility_shocks,
-            variable_info=_mod.variable_info,
+        solve_discrete_problem = get_solve_discrete_problem_value(
+            random_utility_shock_type=internal_model.random_utility_shocks,
+            variable_info=internal_model.variable_info,
             is_last_period=is_last_period,
         )
-        emax_calculators.append(calculator)
+
+        state_choice_spaces[period] = state_choice_space
+        state_space_infos[period] = state_space_info
+        compute_ccv_functions[period] = compute_ccv
+        compute_ccp_functions[period] = compute_ccp
+        solve_discrete_problem_functions[period] = solve_discrete_problem
 
     # ==================================================================================
     # select requested solver and partial arguments into it
     # ==================================================================================
     _solve_model = partial(
         solve,
         state_choice_spaces=state_choice_spaces,
-        continuous_choice_grids=continuous_choice_grids,
         compute_ccv_functions=compute_ccv_functions,
-        emax_calculators=emax_calculators,
+        emax_calculators=solve_discrete_problem_functions,
         logger=logger,
     )
 
+    _next_state_simulate = get_next_state_function(
```

Comment:
Move below `solve_model = ...`. I appreciate having both jit calls next to each other but having a "solve" and a "simulate" block would be even better.